tests/e2e/k8s/test_basic.go

```diff
@@ -14,31 +14,601 @@
-		cl, err := s.fabric.DeployClusterlinks(2, cfg)
+		cl, err := s.fabric.DeployClusterlinks(3, &util.PeerConfig{
+			DataplaneType:           platform.DataplaneTypeEnvoy,
+			Dataplanes:              1,
+			ControlplanePersistency: true,
+		})
 		require.Nil(s.T(), err)
 
-		require.Nil(s.T(), cl[0].CreateExport("echo", &httpEchoService))
-		require.Nil(s.T(), cl[0].CreatePolicy(util.PolicyAllowAll))
-		require.Nil(s.T(), cl[1].CreatePeer(cl[0]))
+		client0 := cl[0].Client()
+		client1 := cl[1].Client()
+
+		// test import API
+		imp := api.Import{
+			Name: "echo",
+			Spec: api.ImportSpec{
+				Service: api.Endpoint{
+					Host: "echo-import",
+					Port: 1234,
+				},
+			},
+		}
+
+		// list imports when empty
+		o, err := client0.Imports.List()
+		require.Nil(s.T(), err)
+		require.Empty(s.T(), o.(*[]api.Import))
+
+		// get non-existing import
+		_, err = client0.Imports.Get(imp.Name)
+		require.NotNil(s.T(), err)
+
+		// delete non-existing import
+		require.NotNil(s.T(), client0.Imports.Delete(imp.Name))
+		// update non-existing import
+		require.NotNil(s.T(), client0.Imports.Update(&imp))
+		// create import
+		require.Nil(s.T(), client0.Imports.Create(&imp))
+		// create import when it already exists
+		require.NotNil(s.T(), client0.Imports.Create(&imp))
 
 		importedService := &util.Service{
-			Name: "echo1",
-			Port: 80,
+			Name: imp.Spec.Service.Host,
+			Port: imp.Spec.Service.Port,
+		}
+
+		accessService := func(allowRetry bool, expectedError error) (string, error) {
+			return cl[0].AccessService(
+				httpecho.GetEchoValue, importedService, allowRetry, expectedError)
 		}
-		require.Nil(s.T(), cl[1].CreateImport("echo", importedService))
 
-		require.Nil(s.T(), cl[1].CreateBinding("echo", cl[0]))
-		require.Nil(s.T(), cl[1].CreatePolicy(util.PolicyAllowAll))
+		// verify import listener is up
+		_, err = accessService(true, &services.ConnectionResetError{})
+		require.ErrorIs(s.T(), err, &services.ConnectionResetError{})
+
+		// get import
+		o, err = client0.Imports.Get(imp.Name)
+		require.Nil(s.T(), err)
+		importFromServer := *o.(*api.Import)
+		require.Equal(s.T(), importFromServer.Name, imp.Name)
+		require.Equal(s.T(), importFromServer.Spec, imp.Spec)
+		require.Equal(s.T(), importFromServer.Status.Listener.Host, "")
+		require.NotZero(s.T(), importFromServer.Status.Listener.Port)
+
+		// list imports
+		o, err = client0.Imports.List()
+		require.Nil(s.T(), err)
+		require.ElementsMatch(s.T(), *o.(*[]api.Import), []api.Import{importFromServer})
+
+		// test binding API
+		binding := api.Binding{
+			Spec: api.BindingSpec{
+				Import: imp.Name,
+				Peer:   cl[1].Name(),
+			},
+		}
+
+		// list bindings when empty
+		o, err = client0.Bindings.List()
+		require.Nil(s.T(), err)
+		require.Empty(s.T(), o.(*[]api.Binding))
+
+		// get non-existing binding
+		_, err = client0.Bindings.Get(binding.Spec.Import)
+		require.Nil(s.T(), err)
+		require.Empty(s.T(), o.(*[]api.Binding))
+
+		// delete non-existing binding
```

Comment:
This code is repeated for binding, import, peer, etc, can you put in a function?